notion-core/src/main/scala/com/ambiata/notion/io/LocationIO.scala

```diff
@@ -48,6 +48,23 @@ case class LocationIO(configuration: Configuration, @transient s3Client: AmazonS
     case h @ HdfsLocation(path)      => Hdfs.isDirectory(new Path(path)).run(configuration)
   }
 
+  def readUTF8(location: Location): ResultTIO[String] = location match {
```

Comment:
I think this should have a consistent set of errors for "File doesn't exist", "File is a directory", at the moment this and almost all the other operations stash failures in RIO which makes them impossible to handle transparently. Location/LocationIO should be be a complete abstraction in error cases as well as happy path.